guides/development/docker-guide.md

```diff
@@ -0,0 +1,241 @@
+# The MarsBased Docker guide for development
+
+At MarsBased we use Docker to work on the development of applications. Using Docker for development has several benefits:
+* Makes setting up the environment for a project a breeze. This dramatically reduces the time to on-board new developers to a project.
+* The environment in which the application runs can be identical to the production environment. This allows to quickly detect bugs dependant on the OS or system packages.
+* Removes the need to have development dependencies (Ruby, Node, PostgreSQL, Redis) installed locally. Only Docker is needed to work on applications.
+* Avoids problems with having multiple versions of dependencies installed. Moreover, stops them for accumulating when versions are updated.
+
+The approach that we follow to configure the Docker environment is heavily inspired by this [magnificent blog post from our evil colleges](https://evilmartians.com/chronicles/ruby-on-whales-docker-for-ruby-rails-development).
+
+The key point of this way working is that **the Docker image is kept as minimal as possible and dependencies are installed on volumes**.
+
+By working this way, the image hardly ever needs to be re-built and we work very closely to how we would do it if we were working with local dependencies. For example: Instead of installing Ruby gems in the image, we mount a volume on the container to store the installed gems and run `bundle install` on the container.
+
+If gems were installed in the image then when there is a change in the used gems (like adding a new gem or updating one), the image needs to be rebuilt, thus needing to install **all gems** again (which can take a long time). By, instead, having gems installed on a volume, when there is a change in a gem, we can just run `bundle install` again (like we would do locally) and it will just install that new gem or updated version.
+
+It is useful to think of it as if the Docker image is just the bare-bones OS and you do the rest the same way you would do it locally (`bundle install`, `npm install`, etc.). More specifically, the image only contains the programming language and system packages (like ImageMagick).
+
+There are 2 options to setup the development environment with Docker:
+* Services only: External services (database, Redis, Minio, Elastic Search, etc.) are run with Docker but the application runs locally.
+* Services + Application: Apart from services, the application is also run in a container.
+
+Running everything with Docker has the advantage of not needing to have any dependency installed locally, apart from Docker. The disadvantage is that it runs slower because the application can't use the full memory + CPU potential from the computer.
+
+When working on a single application it makes sense to use the services only approach, while when working on multiple projects it is more convenient to run all applications in containers to avoid having a dependency hell in the computer.
```

Comment:
"Application" word is confusing (to me). 

```suggestion
When working on a single project, it makes sense to use the services only approach, while when working on several projects it is more convenient to run each inside containers to avoid having a dependency hell in the computer.
```